pkg/hub/controllers/vpnkeyrotation/reconciler.go

```diff
@@ -380,9 +380,15 @@ func (r *Reconciler) syncCurrentRotationState(ctx context.Context,
 		syncedRotationState := make(map[string]hubv1alpha1.StatusOfKeyRotation)
 		for _, gw := range allGwsUnderCluster {
 			obj, ok := currentRotationState[gw]
+
 			if ok {
 				syncedRotationState[gw] = obj
 			} else if !ok {
+				gwIsReady := r.areGatewayPodsReady(ctx, gw)
+				if !gwIsReady {
+					requeue = true
+					return errors.New("gateway pods are not ready")
```

Comment:
@YachikaRalhan you need to requeue also here (bool, error)

Reply:
This is under errors.Retry function which is returning only bool.